### PR TITLE
update split-paired-reads to support -1 and -2 options

### LINKED ISSUE #761: Fix split-paired-reads.py to use khmer.utils.write_record.
The FASTA/FASTQ output code in split-paired-reads should use our new functions for outputting FASTA/FASTQ, which is in khmer/utils.py.

### PR DESCRIPTION
**Pull request by ctb**
Fixes #761, addresses most of https://github.com/ged-lab/khmer/issues/732#issuecomment-71903726 (although streaming won't work until the next release of screed).

- [x] Is it mergable?
- [x] Did it pass the tests?
- [x] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Is it documented in the ChangeLog?
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
@bede @macmanes comments/reviews welcome!

cc @bocajnotnef re streaming (also ref #654)

**Comment by ctb** [comment]
Ready for review @bocajnotnef @luizirber @camillescott @mr-c

**Comment by bede** [comment]
This was the other implementation I had in mind which is why I initially suggested opening an issue. Sounds good! Perhaps it's slightly redundant to have both? But that's not a very good reason.

**Comment by ctb** [comment]
I like all having all of them - a good no-arg default, a good simple "put stuff here", and a good but longer-winded "NO, put stuff HERE".  I can see using all of them myself, too!

**Comment by bede** [comment]
> I can see using all of them myself, too!
Me too!

File: scripts/split-paired-reads.py [hunk 1]
```diff
@@ -34,6 +35,11 @@ def get_parser():
     specified using :option:`-o`/:option:`--output-dir`. This directory will be
     created if it does not already exist.
 
+    Alternatively, you can specify the filenames directly with
+    :option:`-1`/:option:`output_first` and
```
Comment (mr-c) [inline scripts/split-paired-reads.py]: Missing leading double dash before output_first

File: scripts/split-paired-reads.py [hunk 2]
```diff
@@ -78,39 +97,28 @@ def main():
...
-
-    if hasattr(record, 'accuracy'):
-        is_fastq = True
-
     counter1 = 0
     counter2 = 0
     index = None
     for index, record in enumerate(screed.open(infile)):
-        if index % 100000 == 0:
+        if index % 100000 == 0 and index:
```
Comment (mr-c) [inline scripts/split-paired-reads.py]: Why 'and index'?

Comment (ctb) [inline scripts/split-paired-reads.py]: On Wed, Feb 11, 2015 at 02:50:55AM -0800, Michael R. Crusoe wrote:
> >      counter1 = 0
> >      counter2 = 0
> >      index = None
> >      for index, record in enumerate(screed.open(infile)):
> > -        if index % 100000 == 0:
> > +        if index % 100000 == 0 and index:
> 
> Why 'and index'? 

I hate seeing the '0' output.

**Comment by ctb** [comment]
Ready for merge?

**Comment by ctb** [comment]
w00t!

**Pull request merged**